.github/workflows/instascale-release.yml

```diff
@@ -1,37 +1,66 @@
-# This workflow will build an image for the instascale-controller, then push the image to the `project-codeflare` quay repo
+# This workflow will release the InstaScale repository: build an image for the instascale-controller, then push the image to the `project-codeflare` quay repo
 
 name: InstaScale Release
 
 on:
   workflow_dispatch:
-  push:
-    branches: [ "main" ]
-    paths:
-      - "VERSION"
+    inputs:
+      tag:
+        description: 'Release tag'
+        required: true
+        default: 'v0.0.0-dev'
+      mcad-version:
+        description: 'Published version of multi-cluster-app-dispatcher to be used as a dependency for InstaScale'
+        required: true
+        default: 'v0.0.0-dev'
 
 jobs:
   release:
     runs-on: ubuntu-latest
+
+    # Permission required to create a release
+    permissions:
+      contents: write
+
     steps:
     - uses: actions/checkout@v3
 
-    - name: Grab New Version Tag
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: 1.18
+
+    - name: Update to specific version of multi-cluster-app-dispatcher dependency
       run: |
-        echo "tag=$(cat VERSION)" >> $GITHUB_ENV
+        go get github.com/project-codeflare/multi-cluster-app-dispatcher@${{ github.event.inputs.mcad-version }}
```

Comment:
It seems the released artefacts won't reflect the source code in GitHub, which somehow breaks reproducibility of the build/release process?

What are the benefits of doing dependencies upgrade as part of the release process, rather than as part of the normal development process?

Reply:
I mean a version vX.Y.Z of InstaScale can be released using a version of MCAD, but the vX.Y.Z tag in the InstaScale repository won't reflect that version. I think that's a best practice to have the source code as the source of truth, so builds can be reproducible.